public/app/plugins/panel/heatmap/HeatmapPanel.tsx

```diff
@@ -151,9 +153,15 @@ const HeatmapPanelViz = ({
       hideGE: options.filterValues?.ge,
       exemplarColor: options.exemplars?.color ?? 'rgba(255,0,255,0.7)',
       yAxisConfig: options.yAxis,
-      ySizeDivisor: scaleConfig?.type === ScaleDistribution.Log ? +(options.calculation?.yBuckets?.value || 1) : 1,
+      ySizeDivisor:
+        activeScaleConfig?.type === ScaleDistribution.Log || activeScaleConfig?.type === ScaleDistribution.Symlog
+          ? options.rowsFrame?.yBucketScale
+            ? 1
+            : +(options.calculation?.yBuckets?.value || 1)
+          : 1,
```

Comment:
I would pull this to a variable and avoid the nested ternary.

Reply:
Good call, it's a headache for sure!